src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -101,42 +103,48 @@ public function __construct(
 	 *
 	 * @param string $url URL to get.
 	 * @return Response Response for the executed request.
-	 * @throws FailedToGetFromRemoteUrl If retrieving the contents from the URL failed.
+	 * @throws FailedToGetCachedResponse If retrieving the contents from the URL failed.
 	 */
 	public function get( $url ) {
-		$cache_key   = self::TRANSIENT_PREFIX . md5( __CLASS__ . $url );
-		$cached_data = get_transient( $cache_key );
-		$headers     = [];
-		$status      = null;
+		$cache_key       = self::TRANSIENT_PREFIX . md5( __CLASS__ . $url );
+		$cached_response = get_transient( $cache_key );
+		$headers         = [];
 
-		if ( false !== $cached_data ) {
+		if ( false !== $cached_response ) {
 			if ( PHP_MAJOR_VERSION >= 7 ) {
-				$cached_data = unserialize( $cached_data, [ CachedData::class, DateTimeImmutable::class ] ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_unserialize,PHPCompatibility.FunctionUse.NewFunctionParameters.unserialize_optionsFound
+				$cached_response = unserialize( $cached_response, [ CachedResponse::class, DateTimeImmutable::class ] ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_unserialize,PHPCompatibility.FunctionUse.NewFunctionParameters.unserialize_optionsFound
 			} else {
 				// PHP 5.6 does not provide the second $options argument yet.
-				$cached_data = unserialize( $cached_data ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_unserialize
+				$cached_response = unserialize( $cached_response ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_unserialize
 			}
 		}
 
-		if ( false === $cached_data || $cached_data->is_expired() ) {
+		if ( false === $cached_response || $cached_response->is_expired() ) {
 			try {
 				$response = $this->remote_request->get( $url );
 				$status   = $response->getStatusCode();
 				$expiry   = $this->get_expiry_time( $response );
 				$headers  = $response->getHeaders();
 				$body     = $response->getBody();
 			} catch ( FailedToGetFromRemoteUrl $exception ) {
-				$status = $exception->getStatusCode();
-				$expiry = new DateTimeImmutable( "+ {$this->min_expiry} seconds" );
-				$body   = $exception->getMessage();
+				$status   = $exception->getStatusCode();
+				$expiry   = new DateTimeImmutable( "+ {$this->min_expiry} seconds" );
+				$body     = $exception->getMessage();
+				$response = new RemoteGetRequestResponse( $body, $headers, $status );
 			}
 
-			$cached_data = new CachedData( $body, $expiry );
+			$cached_response = new CachedResponse( $body, $headers, $status, $expiry );
 
-			set_transient( $cache_key, serialize( $cached_data ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+			set_transient( $cache_key, serialize( $cached_response ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+
+			return $response;
```

Comment:
Why is this `return` needed?

```suggestion
```

Would removing it make the logic more consistent between cached and non-cached code paths?